### ISSUE TITLE
Funds.fees() "information cannot be scrapped" error

### ISSUE THREAD

**Issue by FFFAR12** [Mael-J/mstarpy#2]
Hello,

I have been encountering this error when executing the funds.fees() function. Whether importing the original package or using a third party package manager like openbb. The exact error outpu I am getting is this one:

`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[8], line 1
----> 1 funds.fees()

File ~\Anaconda3\envs\my_quant_lab\lib\site-packages\mstarpy\funds.py:478, in Funds.fees(self)
    467 def fees(self):
    468     """        
    469     This function retrieves the fees of the fund (by scraping pages);
    470 
   (...)
    476 
    477     """
--> 478     no_site_error(self.code,self.name,self.country,self.site)
    479     result = {}
    480     #headers random agent

File ~\Anaconda3\envs\my_quant_lab\lib\site-packages\mstarpy\error.py:24, in no_site_error(code, name, country, site)
     22     raise ValueError(f"The funds of the country {country} cannot be scraped.")
     23 else:
---> 24     raise ValueError(f"The funds {name} ({code}) cannot be scraped.")

ValueError: The funds (10%) growth in NAV p.a. (F00000UWKD) cannot be scraped.`

And when using openbb is this one:

`
---------------------------------------------------------------------------
ConnectionError                           Traceback (most recent call last)
Cell In[3], line 1
----> 1 fund.fees()

File ~\Anaconda3\envs\my_quant_lab\lib\site-packages\mstarpy\funds.py:484, in Funds.fees(self)
    482 url = f"{self.site}funds/snapshot/snapshot.aspx?id={self.code}&tab=5"
    483 response = requests.get(url, headers=headers, proxies=self.proxies)
--> 484 not_200_response(url,response)
    485 soup = BeautifulSoup(response.text, 'html.parser')
    486 if soup.find(id='managementFeesDiv') == None:

File ~\Anaconda3\envs\my_quant_lab\lib\site-packages\mstarpy\error.py:10, in not_200_response(url, response)
      5 """
      6 This function raise a ConnectionError if the status code a requests is not 200.
      7 
      8 """
      9 if not response.status_code == 200:
---> 10     raise ConnectionError(f"Error {response.status_code} for the api {url}. Message : {response.reason}.")

ConnectionError: Error 404 for the api https://www.morningstar.com/funds/snapshot/snapshot.aspx?id=FOUSA00L8W&tab=5. Message : Not Found.`

I wanted to know if it is something related to the url which is not allowing scrapping or there is non existing content in the page. If not the function may not be working properly.

Really appreciate this wonderful package, thank you very much

**Comment by Mael-J** [comment]
Hello,

It seems that this specific fund  (10%) growth in NAV p.a. (F00000UWKD) is not referenced on MorningStar. 

I cannot find it with the screener tool https://www.morningstar.co.uk/uk/screener/fund.aspx#?filtersSelectedValue=%7B%7D.

I tried other methods and they don't work with the object mstarpy.Funds("F00000UWKD").

Unfortunately, I don't have any control on the referencing of funds by MorningStar, my answer is just a guess.

I would also suggest to use the method feeLevel() to get the fees of a fund. This method use the morningstar API and is more reliable than the method fees() which scrape MorningStar web pages with BeautifulSoup.

Sorry again for this inconvenience.

**Comment by FFFAR12** [comment]
Thank you for the prompt response!

I apologize for not providing a correct example earlier. Let's consider "F00000QGHT" - Vietnam Equity (UCITS) Fund A USD instead as an example. You can find it on the UK Morningstar screen, and then navigate to the 'fees' tab for detailed fee information, as shown here: [https://www.morningstar.co.uk/uk/funds/snapshot/snapshot.aspx?id=F00000QGHT&tab=5]

In the Spanish version of Morningstar, the view of fees is simplified, as seen here: [Morningstar Spain link]

I appreciate your recommendation of the feeLevel() method. While it indeed provides a breakdown of fees relative to other funds, I was curious if the fees() method worked and could get the information on:

**Investment Fees (one off)**
Entry Cost (Maximum Front End Load)	–
Max Entry Cost Acquired	–
Max Exit Charge Field	–
Max Exit Cost Acquired	–
Typical Exit Cost	–
Switching Fee	–
**Investment Fees (Actual)**
Ongoing cost	
Transaction Fee	
Distribution Fee	
Performance Fee	
Management Fee	
**Investment Fees (estimated)**
Ongoing cost	
Transaction Fee	
Distribution Fee	
Performance Fee	
Management Fee

By the way, spanish Morningstar screener looks like it is providing the fees information in a much simpler way: [https://www.morningstar.es/es/funds/snapshot/snapshot.aspx?id=F00000QGHT&tab=5]

Thank you!

**Comment by Mael-J** [comment]
Hello,

I see that there is an API https://lt.morningstar.com/api/rest.svc/klr5zyak8x/security_details to get the fees.

I updated the package both of GitHub and PyPi (version 1.0.2). 

You will be able to find the method feeMifid to get the fees.

>>> mstarpy.Funds("F00000UWKD").feeMifid()

I also added the methods sustanability, snapshot and investmentLookup which rely on the same lt.morningstar.comAPI.

Thank you.
